### PR TITLE
Fixed #147703 - MediaQueryListListener interface definition incorrect

### PR DESCRIPTION
**Pull request by irrationalRock**
This PR addresses this issue: https://github.com/Microsoft/TypeScript/issues/14703

I ported over @mkhazov code to addedTypes.json and overridingTypes.json.

Thanks for considering this PR.

### CODE REVIEW

File: baselines/dom.generated.d.ts [hunk 1]
```diff
@@ -9820,6 +9821,25 @@ declare var MediaQueryList: {
...
+declare var MediaQueryListEvent: {
+    prototype: MediaQueryListEvent;
+    new(init: {matches: boolean, media: string}): MediaQueryListEvent;
+};
+
+interface MediaQueryListListener {
+    (mql: MediaQueryListEvent): void;
+}
+
+declare var MediaQueryListListener: {
```
Comment (mhegazy) [inline baselines/dom.generated.d.ts]: this should not be here. only the type.

Comment (mhegazy) [reply baselines/dom.generated.d.ts]: actually i am not sure why you even have this type.

**Comment by mhegazy** [comment]
@saschanaz can you please review this change

**Comment by saschanaz** [comment]
This type is from [the old 2013 CSSOM View spec](http://www.w3.org/TR/2013/WD-cssom-view-20131217/#mediaquerylistlistener), and [has been replaced](http://www.w3.org/TR/cssom-view/#dom-mediaquerylist-addlistener) with the standard EventListener.

Old:

```webidl
interface MediaQueryList {
  readonly attribute DOMString media;
  readonly attribute boolean matches;
  void addListener(MediaQueryListListener listener);
  void removeListener(MediaQueryListListener listener);
};

callback MediaQueryListListener = void (MediaQueryList list);
```

Latest:

```
interface MediaQueryList : EventTarget {
  readonly attribute DOMString media;
  readonly attribute boolean matches;
  void addListener(EventListener? listener);
  void removeListener(EventListener? listener);
           attribute EventHandler onchange;
};
```

**Comment by saschanaz** [comment]
So my suggestion: remove the type entirely.

**Comment by irrationalRock** [comment]
Thanks for the feedback. 
I removed `MediaQueryListListener` and made `MediaQueryList` extend `EventTarget`.

File: inputfiles/removedTypes.json [hunk 2]
```diff
@@ -24,6 +24,7 @@
     },
     "interfaces": {
         "interface": {
+            "MediaQueryListListener": null,
```
Comment (saschanaz) [inline inputfiles/removedTypes.json]: Can you move this line to below HTMLHRElement and over MSStreamReader?

**Pull request merged**